core/src/test/java/fr/sncf/osrd/envelope_sim_infra/EnvelopeTrainPathTest.java

```diff
@@ -0,0 +1,144 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.collect.ImmutableRangeMap;
+import com.google.common.collect.Range;
+import com.google.common.collect.TreeRangeMap;
+import fr.sncf.osrd.api.FullInfra;
+import fr.sncf.osrd.envelope_sim.electrification.Electrification;
+import fr.sncf.osrd.envelope_sim.electrification.Electrified;
+import fr.sncf.osrd.envelope_sim.electrification.Neutral;
+import fr.sncf.osrd.external_generated_inputs.ElectricalProfileMapping;
+import fr.sncf.osrd.railjson.schema.external_generated_inputs.RJSElectricalProfileSet;
+import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSTrackRange;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import java.util.List;
+import java.util.Map;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EnvelopeTrainPathTest {
+
+    private FullInfra infra;
+
+    @BeforeAll
+    public void setUp() {
+        infra = getSmallInfra();
+    }
+
+    @Test
+    public void envelopeFromPathTestElectrificationMap() {
+        var path = makePath(infra.blockInfra(), infra.rawInfra(), 0);
+        var envelopeSimPath = from(path, null);
+
+        TreeRangeMap<Double, Electrification> expectedElectrificationMapBuild = TreeRangeMap.create();
+        expectedElectrificationMapBuild.put(Range.closedOpen(0.0, 80.0), new Electrified("1500"));
+        expectedElectrificationMapBuild.put(Range.closedOpen(80.0, 130.0), new Neutral(true, new Electrified("1500")));
+        expectedElectrificationMapBuild.put(Range.closedOpen(130.0, 180.0), new Electrified("1500"));
+        expectedElectrificationMapBuild.put(Range.closedOpen(180.0, 1980.0), new Electrified("25000"));
+        ImmutableRangeMap<Double, Electrification> expectedElectrificationMap = ImmutableRangeMap.copyOf(
+                expectedElectrificationMapBuild);
+
+        var electrificationMap = envelopeSimPath.getElectrificationMap(null, null, null, true);
+        assertEquals(expectedElectrificationMap, electrificationMap);
+    }
+
+    @Test
+    public void envelopeFromPathTestAverageGrades() {
+        var path = makePath(infra.blockInfra(), infra.rawInfra(), 10);
+        var envelopeSimPath = from(path, null);
+
+        assertEquals(1600, envelopeSimPath.length);
+        assertEquals(-2.625, envelopeSimPath.getAverageGrade(0, 1600));
+        assertEquals(0, envelopeSimPath.getAverageGrade(0, 600));
+        assertEquals(-3, envelopeSimPath.getAverageGrade(600, 900));
+        assertEquals(-6, envelopeSimPath.getAverageGrade(900, 1300));
+        assertEquals(-3, envelopeSimPath.getAverageGrade(1300, 1600));
+        assertEquals(-4, envelopeSimPath.getAverageGrade(700, 1000));
+    }
+
+    @Test
+    public void envelopeFromPathTestPathNoDeadSection() {
+        var path = makePath(infra.blockInfra(), infra.rawInfra(), 10);
+        from(path, null);
```

Comment:
This doesn't really test anything. If you're just checking for the absence of assert or exception, it could be a comment